### ISSUE TITLE
mean_squared_error accuracy

### ISSUE THREAD

**Issue by firatsarlar** [keras-team/keras#16102]
My currenct observation with "mean_squared_error" loss,
If you use mean_squared_error loss and metrics= ["acc"] ,
during training you can't see any meaningfull accuracy,
I wrote my custom acc. function , it's very hard for me to make a PR, because of project complexty,

Here is my fn.

```
def macc(y_true, y_pred):
    return tf.math.maximum( 1 - tf.math.sqrt(tf.losses.mean_squared_error(y_true,y_pred)),0)
```

And the reason to open this issue is , whenever I need to load this compiled model, I need to use :

`model = tf.keras.models.load_model(model_name, custom_objects={"macc": macc })
`

This additional code lines could save some one else,
or I may be wrong some where on the road , and issue turns question ?

( by the way, I'm at very begining of my ML experience, and this issue would be meaningless , please lead me to right direction if I'm wrong)

I use one of my colab experiments to reproduce issue
This is colab link
https://colab.research.google.com/drive/1hYG4RmPaUlnEcRm28NE6Y3HxIMoq1DTM?usp=sharing

These are the cells related the issue :

https://colab.research.google.com/drive/1hYG4RmPaUlnEcRm28NE6Y3HxIMoq1DTM#scrollTo=lL3kJjI-cXLS&line=4&uniqifier=1

https://colab.research.google.com/drive/1hYG4RmPaUlnEcRm28NE6Y3HxIMoq1DTM#scrollTo=eYb8YEiRvVuK&line=3&uniqifier=1

Best

**Comment by Cheril311** [comment]
@firatsarlar the formal definition of accuracy is number of <b> no. of true prediction / total no. of predictions </b> which is how it is implemented in keras, in your use case the number of true predictions will be nearly 0 as it is a regression problem thus you're getting the weird results, as accuracy was designed to be a classification metric not a regression metric in the first place. You can always use simple RMSE for regression instead of how you've used, lower the RMSE on test set, better your model

**Comment by firatsarlar** [comment]
I will do try RMSE with accuracy. As I said I'm very beginning of road of ML.
It may more appropriate after testing RMSE, but formal definition, or results seemed weird to me. 
( I mean , currently I'm disagree with you )
But you probably know better.
Feel free to close the issue.

**Comment by firatsarlar** [comment]
I tried RMSE, but new question is it is not exists as loss, it is a metric.
```
    model.compile(optimizer=tf.keras.optimizers.Adam(learning_rate=0.01),
                        
                        loss=[tf.losses.mean_squared_error],
                        
                        metrics=[mean_squared_error_acc if useCustomAcc else tf.keras.metrics.RootMeanSquaredError()]
                )

```
 (I have updated the notebook , link changed )
https://colab.research.google.com/drive/1hYG4RmPaUlnEcRm28NE6Y3HxIMoq1DTM?usp=sharing

I still can't see accuracy as percentage ( I mean in any format 0.50 or 50  )
Still I may be wrong, but I think this is an issue not a question.

By the way, I personally got what I wanted, but my intention is to contribute.

**Comment by jvishnuvardhan** [comment]
@firatsarlar As @Cheril311 mentioned, `accuracy` is a metric designed for classification. Your custom metric is simply max(1-`mse`,0). What you are estimating through your custom metric is (1-error_between_y_true_and_y_pred).

Based on your use case and domain knowledge, You can define a custom metric saying that if the error is <x, accuracy is 1 and then find mean accuracy of all samples to find the accuracy in the prediction.

Another suggestion is you can reduce your `target` in your `EarlyStoppingAtMinLoss` code to improve the model performance. In [this](https://colab.research.google.com/gist/jvishnuvardhan/2bd5c3745e4662ae4ee773ea2ea93e26/keras_metricissue_repr.ipynb) attached gist I updated `target` to get better model performance. Thanks!

**Comment by firatsarlar** [comment]
I was intentionally keeping target to make things fast ( not to waste someone's time ) and it's currently out of context.
I still trying to tell accuracy is what as it is.  Formal definitions, current dominant knowledge,
 may suggest society ( scientists ) to define it as you say.
 Although I'll close the issue , I'm not enlightened about this topic.
 Wish me luck.

**Comment by firatsarlar** [comment]
https://colab.research.google.com/drive/1hYG4RmPaUlnEcRm28NE6Y3HxIMoq1DTM?usp=sharing

I decided to reopen this as a feature request and proof of concept could be the link I've shared.
The line and colab notebook ( it will ask you to sign in to google even I chose to anyone with the lik ) 

It's about to intuitiveness.
Giving a target like : 0.9
instead of 
target like : 1e-7
is reasonably lower the requirement of current terms of ML  -even scientific notation is an extra for an elementary kid-, 

Feel free to close issue. ( Again, I'm trying to contribute to the community )
